### DISCUSSION TITLE
The future of this theme

### DISCUSSION THREAD

**Discussion by CaiJimmy** [CaiJimmy/hugo-theme-stack#630] [General]
In one month this theme will be three years old. It has become one of Hugo's popular themes, something that I never expected before its release, reaching 2.2K stars so far, and almost 70 contributors. Thanks to all of you to make this possible.

Now, I haven't been actively participating in this community for a long time (maybe since the beginning of my colleague year). Big thanks to all contributors who have been helping responding issues and questions.

As I said before, I didn't expect this theme to be widely used (was my blog theme). And after these years, **I'm not satisfied with the codebase, and I feel necessary for a major rewrite.** So, from now, **I'll stop accepting feature requests and PR until a new major version is released (if it ever comes out).**

I'll try to get a to-do list of things that I have in mind, and if any contributor wants to collaborate please ask ❤. I thought that I would have more spare time in summer but it doesn't seem so (looking for internships and etc.), so any help is appreciated.

**Comment by CaiJimmy** [comment]
Note: I closed my blog for a long time and stopped using this theme. The major rewrite is aimed at better maintainability, just in case someone would like to take the baton.

**Reply by tesujimath** [reply]
Thank you for sharing the theme as you did. Many of us have really appreciated your work. ♥️

**Reply by axelsegebrecht** [reply]
I second what @tesujimath said :) I am just embarking on a major rework of my own website and blog and would love to use this theme.

What kind of support are you looking for to rewrite it or maintain the current one?

**Reply by CaiJimmy** [reply]
> What kind of support are you looking for to rewrite it or maintain the current one?

I'm not sure yet. I wanted to get a to-do list of things, but due to some plan changes, I won't be able to work on this until mid-July (or longer).

**Comment by lin-snow** [comment]
thanks

**Comment by CaiJimmy** [comment]
Just a brief update regarding this topic. 

I found a summer internship job and have no more time to spend on this project. Given the current situation, this theme probably will stay as it is right now. I'll continue receiving issue and discussion notifications and respond if I can. **But don't expect any significant changes to come in the future**. After all, it works well right now, better not to break it.
